src/Service/PackageScanner/SensioLabPackageScanner.php

```diff
@@ -0,0 +1,151 @@
+            );
+        } catch (\Throwable $exception) {
+            $result[get_class($exception)] = $exception->getMessage();
+            $this->saveResult($package, ScanResult::STATUS_ERROR, $result);
+
+            return;
+        }
+
+        if ($result === []) {
+            $this->saveResult($package, ScanResult::STATUS_OK, $result);
+
+            return;
+        }
+
+        $this->saveResult($package, ScanResult::STATUS_WARNING, $result);
+    }
+
+    /**
+     * @param mixed[] $result
+     */
+    private function saveResult(Package $package, string $status, array $result): void
+    {
+        $this->results->add(
+            new ScanResult(
+                Uuid::uuid4(),
+                $package,
+                new \DateTimeImmutable(),
+                $status,
+                $result
+            )
+        );
+    }
+
+    private function findDistribution(Package $package): string
+    {
+        $packageName = $package->name();
+        $latestReleasedVersion = $package->latestReleasedVersion();
+
+        $normalizedVersion = $latestReleasedVersion === 'no stable release' ?
+            '9999999-dev' : $this->versionParser->normalize((string) $latestReleasedVersion);
+
+        $providerData = $this->packageManager->findProviders(
+            $package->organizationAlias(),
+            [new PackageName($package->id()->toString(), (string) $package->name())]
+        );
+
+        foreach ($providerData[$packageName] ?? [] as $packageData) {
+            $packageVersion = $packageData['version_normalized'] ?? $this->versionParser->normalize($packageData['version']);
+            $packageDist = $packageData['dist'];
+            $reference = $packageDist['reference'];
+
+            if ($packageVersion === $normalizedVersion && isset($packageDist['url'], $reference)) {
+                $archiveType = $packageDist['type'];
+                $filename = $this->packageManager->distFilename(
+                    $package->organizationAlias(),
+                    (string) $packageName,
+                    $normalizedVersion,
+                    $reference,
+                    $archiveType
+                );
+
+                return $filename->getOrElseThrow(
+                    new \RuntimeException('Distribution file not found')
+                );
+            }
+        }
+
+        throw new \RuntimeException("Version $normalizedVersion for package $packageName not found");
+    }
+
+    private function extractLockFile(string $distFilename): string
+    {
+        $zip = new \ZipArchive();
+        $result = $zip->open($distFilename);
+        if ($result !== true) {
+            throw new \RuntimeException("Error while opening ZIP file '$distFilename', code: $result");
+        }
+
+        $content = null;
+        for ($i = 0; $i < $zip->numFiles; ++$i) {
+            if (preg_match('/\/composer.lock$/', (string) $zip->getNameIndex($i)) === 1) {
```

Comment:
Will this only work for files in the root directory? Because there may be more composer.lock files there